src/LedgerId.js

```diff
@@ -0,0 +1,123 @@
+import * as hex from "./encoding/hex.js";
+
+const DEFAULT_ERROR = "Default case reached for: ";
+
+export default class LedgerId {
+    /**
+     * @hideconstructor
+     * @internal
+     * @param {Uint8Array} ledgerId
+     */
+    constructor(ledgerId) {
+        /**
+         * @readonly
+         * @type {Uint8Array}
+         */
+        this._ledgerId = ledgerId;
+
+        Object.freeze(this);
+    }
+
+    /**
+     * @returns {Uint8Array}
+     */
+    get ledgerId() {
+        return this._ledgerId;
+    }
+
+    /**
+     * @param {string} ledgerId
+     * @returns {LedgerId}
+     */
+    static fromString(ledgerId) {
+        ledgerId =
+            hex.decode(ledgerId).toString().length >= 1
```

Comment:
Probably should put `hex.decode(ledgerId)` into a separate variable since you're doing this twice. I haven't read through your tests yet, but this looks like it'd throw an error if the string is `mainnet`.

Reply:
it doesnt error. its also memory vs time thing. since both are miniscule i went with making the call twice instead of making a new var.

Reply:
Not necessarily for performance reasons, but for code cleanness.